gcloud-java-dns/src/main/java/com/google/cloud/dns/DnsBatch.java

```diff
@@ -337,17 +343,29 @@ public void onFailure(GoogleJsonError googleJsonError) {
    * A joint callback for both "get change request" and "create change request" operations.
    */
   private RpcBatch.Callback<Change> createChangeRequestCallback(final String zoneName,
-      final DnsBatchResult<ChangeRequest> result, final boolean idempotent) {
+      final DnsBatchResult<ChangeRequest> result, final boolean nullForNotFound,
+      final boolean idempotent) {
     return new RpcBatch.Callback<Change>() {
       @Override
       public void onSuccess(Change response) {
-        result.success(response == null ? null : ChangeRequest.fromPb(options.service(),
-            zoneName, response));
+        result.success(response == null ? null : ChangeRequest.fromPb(options.service(), zoneName,
+            response));
       }
 
       @Override
       public void onFailure(GoogleJsonError googleJsonError) {
-        result.error(new DnsException(googleJsonError, idempotent));
+        DnsException serviceException = new DnsException(googleJsonError, idempotent);
+        if (nullForNotFound && serviceException.code() == HTTP_NOT_FOUND) {
+          if ("entity.parameters.changeId".equals(serviceException.location())
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [These nested if statements could be combined](https://www.codacy.com/app/mziccard/gcloud-java/file/2448645437/issues/source?bid=3123103&fileBranchId=3247517#l359)